packages/docusaurus-plugin-openapi/src/index.ts

```diff
@@ -266,30 +284,7 @@ export default function pluginOpenAPI(
             {
               test: /(\.mdx?)$/,
               include: [dataDir, contentPath].map(addTrailingPathSeparator),
-              use: [
-                getJSLoader({ isServer }),
-                {
-                  loader: require.resolve("@docusaurus/mdx-loader"),
-                  options: {
-                    admonitions: options.admonitions,
-                    remarkPlugins,
-                    rehypePlugins,
-                    beforeDefaultRehypePlugins,
-                    beforeDefaultRemarkPlugins,
-                    markdownConfig: siteConfig.markdown ?? { mdx1Compat: {} },
-                    metadataPath: (mdxPath: string) => {
-                      if (mdxPath.startsWith(dataDir)) {
-                        // The MDX file already lives in `dataDir`: this is an OpenAPI MDX
-                        return mdxPath.replace(/(-content\.mdx?)$/, ".json");
-                      } else {
-                        // Standard resolution
-                        const aliasedSource = aliasedSitePath(mdxPath, siteDir);
-                        return join(dataDir, `${docuHash(aliasedSource)}.json`);
-                      }
-                    },
-                  },
-                },
-              ].filter(Boolean),
+              use: [getJSLoader({ isServer }), mdxLoaderItem].filter(Boolean),
```

Comment:
The original error happened because the OpenAPI plugin wasn't passing the `siteDir` option to `mdx-loader`. Apparently that wasn't being used before Docusaurus 3.6 so the plugin still worked, but now it's actually being used.

Reply:
Hmmm I see thanks.

`siteDir` is a required loader option before v3.6, but maybe it wasn't failing before. Unfortunately since you don't use the types you didn't know your code didn't typecheck.

I think it could actually be better if you use `createMDXLoaderItem` to ensure that no attribute is missing if we add more of them in the future. Unfortunately we can't guarantee this won't happen in a minor version 😅 

I think in the future we'll provide some kind of public API provided by the plugin lifecycle such as `getMDXLoader()`, similar to what you already used with `getJSLoader`. 

If we introduce this new API, we may still keep `createMDXLoaderItem()` so it should be fine to use it. You already use non-public APIs anyway 😅